mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2621,6 +2621,15 @@ Type SubViewOp::inferResultType(MemRefType sourceMemRefType,
   dispatchIndexOpFoldResults(offsets, dynamicOffsets, staticOffsets);
   dispatchIndexOpFoldResults(sizes, dynamicSizes, staticSizes);
   dispatchIndexOpFoldResults(strides, dynamicStrides, staticStrides);
+
+  for (int64_t offset : staticOffsets) {
```

Comment:
nit: loops can be wrapped in `#ifndef NDEBUG`

Reply:
Because loops are relatively expensive you mean? Is this comment outdated when the logic moves to the interface and is specified as an invariant?